staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -1691,6 +1694,38 @@ type CSIPersistentVolumeSource struct {
 	NodePublishSecretRef *SecretReference `json:"nodePublishSecretRef,omitempty" protobuf:"bytes,8,opt,name=nodePublishSecretRef"`
 }
 
+// Represents a source location of a volume to mount, managed by an external CSI driver
+type CSIVolumeSource struct {
+	// Driver is the name of the driver to use for this volume.
+	// Required.
+	Driver string `json:"driver" protobuf:"bytes,1,opt,name=driver"`
+
+	// Optional: The value to pass to ControllerPublishVolumeRequest.
```

Comment:
It says optional below - no need to repeat yourself repeat yourself

Reply:
I thought we were not going to support controller-side anything in inline CSI?

Reply:
This should be NodePublish.  The readOnly flag passed to NodePublish comes from the volumeSource.readOnly flag (same as in-tree drivers too).  Although, it should be noted that bind mounting a volume as readonly [doesn't mount propagate to the host](https://github.com/kubernetes/kubernetes/issues/70549#issuecomment-435452402), so maybe this flag is slightly redundant to the volumeMount.readOnly flag.  The only benefit is that you can set readOnly once for the entire volume instead of each container mount.

Reply:
You are right no controller-side anything. Missed comment. Will fix.